core/README.md

```diff
@@ -0,0 +1,160 @@
+# `@commercetools-test-data/core`
+
+This package provides the core functions to define the data models.
+
+## Install
+
+```bash
+$ yarn add -D @commercetools-test-data/core
+```
+
+## Creating a new data model
+
+Models are treated packages. Each model will be published as a standalone NPM package.
+
+To create a new model, add a new package in the `models` workspace with the following structure:
+
+```
+models/<package_name>
+├── LICENSE
+├── README.md
+├── index.ts
+├── package.json
+├── src
+│   ├── builder.ts
+│   ├── generator.ts
+│   ├── index.ts
+│   ├── presets
+│   │   └── index.ts
+│   ├── transformers.ts
+│   └── types.ts
+├── tsconfig.declarations.json
+└── tsconfig.json
+```
+
+To define and build new models, you need to use the core package `@commercetools-test-data/core`.
+
+### Builder
+
+A builder is the result of the data model. Builders allow composition with other models and to transform the data into different shapes, for example `rest` or `graphql`.
+
+When you want to get the generated data, you would **build** the model (builder).
+
+Builders are essentially composed by a generator and (optionally) transformers.
+
+### Generator
+
+A generator is what describes the initial model shape. Most of the fields can and should be initialized with some values. You can define random values using `fake` or static values.
+
+```ts
+const generator = Generator<TAuthor>({
+  name: 'Author',
+  fields: {
+    firstName: fake((f) => f.name.firstName()),
+    locale: 'en',
+  },
+});
+```
+
+> The shape of the generator model is what usually is used in the UI and does not need transformation.
+
+Models can have nested fields, like object or arrays. Most of the time those are also defined as models.
+In this case, the generator can define the value of the nested field as `null`. Those nested models are then built in the builder and or the transformers.
+
+For example, the author model can have a list of books:
+
+```ts
+const generator = Generator<TAuthor>({
+  name: 'Author',
+  fields: {
+    firstName: fake((f) => f.name.firstName()),
+    locale: 'en',
+    books: null,
+  },
+});
+```
+
+Books are defined as another model:
+
+```ts
+const generator = Generator<TBook>({
+  name: 'Book',
+  fields: {
+    title: fake((f) => f.random.words()),
+  },
+});
+```
+
+When defining the builder for the `Author`, we can then initialize the `books` field by referencing the `Book` model.
+
+```ts
+const Author = Builder<TAuthor>({
+  generator,
+  transformers,
+}).books([Book.random()]);
+```
+
+At the same time, we also need to tell the transformers that the `books` field can be built. We do that by specifying the field in the `buildFields` option:
+
+```ts
+const transformers = {
+  default: Transformer<TAuthor, TAuthor>('default', {
+    buildFields: ['books'],
+  }),
+  graphql: Transformer<TAuthor, TAuthorGraphql>('graphql', {
+    buildFields: ['books'],
+    addFields: () => ({
+      __typename: 'Author',
+    }),
```

Comment:
Irrelevant.

```suggestion
```

Reply:
Why is it irrelevant? 🤔

Reply:
The emphasis here is on `buildFields`, adding other stuff will make the reader destructed, in my opinion.

Reply:
I see. I'll put a `// ...` comment then to hint that other fields can be provided.